doc/design/model_zoo_pai.md

```diff
@@ -56,27 +56,78 @@ PAI requires a *PAI program* to expose several command line options to communica
 
 ### Versioning and Releasing
 
-Versioning and releasing in PAI model zoo is the same as what's described in [Model Zoo Design](model_zoo.md#Versioning-and-Releasing). The only requirement is that the base docker image of the SQLFlow model zoo should incorporate both `odpscmd`(which is already in place) and `pyodps`.  We propose to use `pyodps` at the moment because it is easier to be integrated into the new workflow framework that is in progress.
+Versioning and releasing in PAI model zoo is the same as what's described in [Model Zoo Design](model_zoo.md#Versioning-and-Releasing). The only requirement is that the base docker image of the SQLFlow model zoo should incorporate both `odpscmd`(which is already in place) and `pyodps`. We propose to use `pyodps` at the moment because it is easier to be integrated into the new workflow framework that is in progress.
 
 ### Submitter Programs of PAI Model Zoo
 
-Currently, each deployment of SQLFlow has been configured to use only one submitter. So we assume that all the tasks of the deployment of SQLFlow on PAI will be submitted to PAI.
+A [PAI program](#Concepts) of PAI model zoo may have multiple python source files and will be executed as a [PAI task](#Concepts) on PAI rather than a process in a Docker container. Take this into consideration, we propose to impose a requirement on a legal model zoo Docker image: the Docker image should specify the directory where the model source files reside, e.g., in an environment variable `SQLFLOW_MODEL_ZOO_MODEL_SOURCE`. In that case, whether or not a model is executed on PAI, its [submitter program](model_zoo.md#Submitter-Programs) can easily import the *model definition*. For example, suppose we have an environment variable `SQLFLOW_MODEL_ZOO_MODEL_SOURCE=/model_zoo/` in a model zoo Docker image and the *model definition* is a python class `MyAwesomeModel` defined in a python script `/model_zoo/my_awesome_model.py`:
```

Comment:
> A [PAI program](#Concepts) of PAI model zoo may have multiple python source files and will be executed as a [PAI task](#Concepts) on PAI rather than a process in a Docker container.

maybe just say: PAI do not support run the model in a Docker container.

Reply:
> the Docker image should specify the directory where the model source files reside

A Docker image can not specify anything, the model developer (data scientists) can. Or we can say: "The source files of the model must be put into the directory `/workspace` in the Docker image"

Reply:
> `SQLFLOW_MODEL_ZOO_MODEL_SOURCE=/model_zoo/`

the directory where to put model source files is not the "model zoo" I think. It may contain several models, but not **the** "model zoo".

Reply:
> > `SQLFLOW_MODEL_ZOO_MODEL_SOURCE=/model_zoo/`
> 
> the directory where to put model source files is not the "model zoo" I think. It may contain several models, but not **the** "model zoo".

`model_zoo` is only an example here, it's the root of all the models in that Docker image.

Reply:
> model_zoo is only an example here, it's the root of all the models in that Docker image.